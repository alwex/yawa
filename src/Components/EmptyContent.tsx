import React from 'react'
import { View, Text, Icon } from 'native-base'
import style from './Styles/EmptyContentStyle'
import IconType from '../Types/IconType'

interface EmptyContentProps {
  iconName: string
  iconType: IconType
  text: string
}

export default class EmptyContent extends React.Component<EmptyContentProps> {
  render() {
    const { iconType, iconName, text } = this.props
    return (
      <View style={style.container}>
        <View style={style.row}>
          <Icon style={style.icon} name={iconName} type={iconType} />
        </View>
        <View>
          <Text style={style.description}>{text}</Text>
        </View>
      </View>
    )
  }
}
