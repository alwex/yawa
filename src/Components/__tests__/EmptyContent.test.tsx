import React from 'react'
import renderer from 'react-test-renderer'
import EmptyContent from '../EmptyContent'

describe('EmptyContent', () => {
  it('render correctly', () => {
    const component = renderer
      .create(<EmptyContent text="hello" iconName="account" iconType="MaterialCommunityIcons" />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
