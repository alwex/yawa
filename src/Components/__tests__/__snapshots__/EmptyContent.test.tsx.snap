// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyContent render correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": "50%",
        "textAlign": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "height": 200,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#CECDD2",
              "fontSize": 100,
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      type="MaterialCommunityIcons"
    >
      ï€„
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#CECDD2",
            "fontSize": 20,
          },
        ]
      }
      uppercase={false}
    >
      hello
    </Text>
  </View>
</View>
`;
